contracts/utils/VaultQueue.sol

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.4;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import {
+    ReentrancyGuardUpgradeable
+} from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {ISAVAX} from "../interfaces/ISAVAX.sol";
+import {IRibbonVault} from "../interfaces/IRibbon.sol";
+import {IRibbonThetaVault} from "../interfaces/IRibbonThetaVault.sol";
+import {Vault} from "../libraries/Vault.sol";
+import {DSMath} from "../vendor/DSMath.sol";
+
+interface IVaultQueue {
+    struct Transfer {
+        address creditor;
+        address srcVault;
+        address dstVault;
+        uint256 amount;
+        uint256 timestamp;
+    }
+}
+
+contract VaultQueue is ReentrancyGuardUpgradeable, Ownable {
+    mapping(address => IVaultQueue.Transfer[]) public qTransfer;
```

Comment:
I think it may make sense to fix the Transfer[] array size to a number so we can bound the number of transfers.

Reply:
Instead of fixing the array size, we could limit the queue array by checking `require(qTransfer.length < 256);` in the queuing function.